### PR TITLE
Enable FP8 quantization in SDXL using INC

### PR DESCRIPTION
**Pull request by splotnikv**
# What does this PR do?
This PR illustrates how to enable FP8 quantization in SDXL using INC https://docs.habana.ai/en/latest/PyTorch/Inference_on_PyTorch/Inference_Using_FP8.html

### CODE REVIEW

File: examples/stable-diffusion/text_to_image_generation.py [hunk 1]
```diff
@@ -270,6 +270,12 @@ def main():
         action="store_true",
         help="Enable deterministic generation using CPU Generator",
     )
+    parser.add_argument(
+        "--quant_mode",
+        default="disable",
+        type=str,
+        help="Quantization mode 'measure', 'quantize' or 'disable'",
+    )
```
Comment (libinta) [inline examples/stable-diffusion/text_to_image_generation.py:278]: can you do the same as run_generation? and not introduce extra arguments?

File: optimum/habana/diffusers/pipelines/stable_diffusion_xl/pipeline_stable_diffusion_xl.py [hunk 2]
```diff
@@ -479,6 +479,22 @@ def __call__(
             `tuple`. When returning a tuple, the first element is a list with the generated images.
         """
 
+        quant_mode=kwargs["quant_mode"]
+        if quant_mode == "measure" or quant_mode == "quantize":
```
Comment (libinta) [inline optimum/habana/diffusers/pipelines/stable_diffusion_xl/pipeline_stable_diffusion_xl.py:483]: please check how run_generation is done

File: optimum/habana/diffusers/pipelines/stable_diffusion_xl/pipeline_stable_diffusion_xl.py [hunk 3]
```diff
@@ -479,6 +479,22 @@ def __call__(
...
+
+            import habana_frameworks.torch.core as htcore
+            htcore.hpu_set_env()
+
+            from neural_compressor.torch.quantization import FP8Config, convert, prepare
+            config = FP8Config.from_json_file(quant_config_path)
+            if config.measure:
+                self.unet = prepare(self.unet, config)
+            elif config.quantize:
+                self.unet = convert(self.unet, config)
```
Comment (kumarans-ai) [inline optimum/habana/diffusers/pipelines/stable_diffusion_xl/pipeline_stable_diffusion_xl.py:495]: All Unet steps should not be converted to Fp8, Recipe for good accuracy includes last few steps in bf16. So, you must have the bf16 Unet support - refer Model Garden reference for Fp8-INC conversion patch

**Review by kumarans-ai — CHANGES_REQUESTED** [review]

**Comment by libinta** [comment]
@splotnikv can you update the branch ? thanks

**Comment by splotnikv** [comment]
I have not forgot about this PR but too busy right now to address feedback. I’ll try to push update next week.

File: examples/stable-diffusion/text_to_image_generation.py (hunk 1 above)
Comment (splotnikv) [reply examples/stable-diffusion/text_to_image_generation.py]: done

File: optimum/habana/diffusers/pipelines/stable_diffusion_xl/pipeline_stable_diffusion_xl.py (hunk 2 above)
Comment (splotnikv) [reply optimum/habana/diffusers/pipelines/stable_diffusion_xl/pipeline_stable_diffusion_xl.py]: done

File: optimum/habana/diffusers/pipelines/stable_diffusion_xl/pipeline_stable_diffusion_xl.py (hunk 3 above)
Comment (splotnikv) [reply optimum/habana/diffusers/pipelines/stable_diffusion_xl/pipeline_stable_diffusion_xl.py:496]: ok, I'll submit it as a separate PR. Original intention of this PR is to show how easy it is to quantize model, so I prefer to keep code change to bare minimum.

**Comment by libinta** [comment]
@splotnikv if you can't finish it this week, we won't be able to pull in the next release.

**Comment by splotnikv** [comment]
> @splotnikv if you can't finish it this week, we won't be able to pull in the next release.

I don't have anything else to change in this PR. From my point of view it is ready. And I also don't mind if it'll go into next release.

**Comment by kumarans-ai** [comment]
@splotnikv, as part of release commit for 1.19.0, we have already completed this feature plan to merge this below PR for Fp8 flow https://github.com/huggingface/optimum-habana/pull/1519.

**Comment by libinta** [comment]
@splotnikv  We will merge 1519 that has fp8 support

**Pull request closed without merging**